qubes/vm/mix/dvmtemplate.py

```diff
@@ -293,12 +301,54 @@ def get_feat_preload(self) -> list[str]:
         value = self.features.get(feature, "")
         return value.split(" ") if value else []
 
-    def get_feat_preload_max(self) -> int:
-        """Get the ``preload-dispvm-max`` feature as an integer."""
+    def get_feat_global_preload_max(self) -> Optional[int]:
+        """Get the global ``preload-dispvm-max`` feature as an integer if it is
+        set, None otherwise."""
+        feature = "preload-dispvm-max"
+        assert isinstance(self, qubes.vm.BaseVM)
+        value = None
+        global_features = self.app.domains["dom0"].features
+        if feature in global_features:
+            value = int(global_features.get(feature) or 0)
+        return value
+
+    def get_feat_preload_max(self, force_local=False) -> int:
+        """Get the ``preload-dispvm-max`` feature as an integer.
+
+        :param global_feat: ignore global setting.
+        """
         feature = "preload-dispvm-max"
         assert isinstance(self, qubes.vm.BaseVM)
-        value = self.features.get(feature, 0)
-        return int(value) if value else 0
+        value = None
+        if (
+            not force_local
+            and not getattr(self, "preload_max_ignore_global", False)
+            and self == getattr(self.app, "default_dispvm", None)
+        ):
+            value = self.get_feat_global_preload_max()
+        if value is None:
+            value = self.features.get(feature)
+        return int(value or 0)
+
+    def is_global_preload_set(self) -> bool:
+        """Returns ``True`` if this qube is the global default_dispvm and the
+        global preload feature is set."""
+        assert isinstance(self, qubes.vm.BaseVM)
+        if (
+            self == getattr(self, "default_dispvm", None)
```

Comment:
```suggestion
            self == getattr(self.app, "default_dispvm", None)
```
otherwise you'll check VM's "default_dispvm" property, not the global one